AsyncDisplayKit/ASDisplayNode.mm

```diff
@@ -1645,17 +1647,12 @@ - (void)recursivelyClearFetchedData
 }
 
 /**
- * We currently only set interface state on nodes
- * in table/collection views. For other nodes, if they are
- * in the hierarchy we return `Unknown`, otherwise we return `None`.
- *
- * TODO: Avoid traversing up node hierarchy due to possible deadlock.
- * @see https://github.com/facebook/AsyncDisplayKit/issues/900
- * Possible solution is to push `isInCellNode` state downward on `addSubnode`/`removeFromSupernode`.
+ * We currently only set interface state on nodes in table/collection views. For other nodes, if they are
+ * in the hierarchy we enable all ASInterfaceState types with `ASInterfaceStateInHierarchy`, otherwise `None`.
  */
-- (BOOL)supportsInterfaceState {
-  return ([self isKindOfClass:ASCellNode.class]
-      || [self _supernodeWithClass:ASCellNode.class checkViewHierarchy:NO] != nil);
+- (BOOL)supportsRangeManagedInterfaceState
```

Comment:
@levi I think this will remain, although the name is probably unclear and needs to be cleaned up.  It will always be possible to add a node directly to a UIWindow or other non-ASDK hierarchy, and it seems to me like we'd either have to require developers to manually setInterfaceState: for even basic functionality or have something like this to enable some stuff automatically.

I may have missed your point and want to make sure to understand, because this area is not fully thought-through and your intuition / suggestions could be key.

For what it's worth, I do eventually want to enable interfaceState as a settable property (I think it is readonly right now) so that advanced custom users can set the different states manually.  Both to implement their own range controller-like patterns (hello, Paper!), but also simple to prompt a hierarchy to do async measurement or display ahead of insertion in the hierarchy.